### ISSUE TITLE
Feature suggestion: disable the new Up Next function

### ISSUE THREAD

**Issue by ackern** [YePpHa/YouTubeCenter#1591]
Youtube added a new piece of shitty bullshit in recent weeks in yet another attempt to force us to watch what they tell us to watch. It has the effect of shutting down our ability to replay a video we just watched. 

So, if you had the time and found a way to disable Up Next, that would be great.

Basically, Up Next is  a new feature that jumps to some other video suggested by Youtube when your video's done, and there's no way to stop it. Which means now the only way to replay what you watched is to use the browser Back button and reload the entire page. 

Furthermore it means if you wanted to view some videos that showed up in the right-hand column, those will all be different now so for this too you need to browser-Back and hope they're still listed.

Thanks again for the amazing YTC. My head explodes every time I have to watch youtube without it.

**Comment by Zarggg** [comment]
You can disable Up Next entirely by flipping the toggle in the upper-right corner (the color will also change from blue to grey). This works on both Firefox and Chrome; I haven't tested other browsers.

Screenshot for reference: http://i.imgur.com/Krz4pvV.png

**Comment by z235** [comment]
And the next time you visit YouTube without stored cookies it will be activated again.

I would be very grateful if you could add a toggle somewhere in YTC. I don't want to be required to save my cookies forever for this site to turn this annoying feature off.

**Comment by ackern** [comment]
>You can disable autoplay on Up Next by flipping the toggle in the upper-right corner

That's incorrect. On my computer flipping the toggle makes no difference. I already tried that and it jumps to the Up Next anyway. It does not start *playing* that next video (because I have YTC set to block autoplay) but you do get sent to the page of that video and the video loads. So AFAIK when you have YTC set up the way I do, the toggle has no effect of any kind.

**Comment by ErdeFB** [comment]
@ackern does the setting get reset when you have YTC disabled? If not, it is not a YTC problem. Would be a nice feature though for all the people who are cookie paranoid.

**Comment by ackern** [comment]
I don't understand your question. There's too many "not"s.  When YTC is disabled, the toggle functions correctly for a limited period of time. I have noticed  unpredictable variations in how long it 'sticks'. Sometimes when I switch it off,  Youtube switches it back on after a video or two and sometimes it seems to stay off the whole day, or anything in between. But I don't precisely test because I can't stand to watch that much with YTC turned off.

**Comment by LittleVulpix** [comment]
Duplicate of #1581 , I agree that it is very annoying and I have the same experience with the "disable autoplay" thing not helping at all.

**Comment by ErdeFB** [comment]
Disable autoplay is not supposed to, if I recall, to stop the up next video switch from happening.

Anyway, I don't believe it is YTC that messes with the up next toggle getting reset. It would, however, be beneficial for having a option to always check the status of the toggle and then switch it on or off as preferred.

**Comment by Zarggg** [comment]
The state of the Up Next Autoplay switch is stored by a YouTube site cookie. If you are deleting cookies set by YouTube and are not using an account, then you should expect settings to reset to their default state the next time you load the site.

**Comment by YotaXP** [comment]
I had this problem.  The switch was in the "Off" state, yet Up Next was still triggering.  Flipping it on and back off didn't help.  Recently however, I tried once again flipping it on and back off, and it appears to have finally stopped.  The problem may have been partially fixed.  Has anyone else tried toggling it recently?
![Autoplay Toggle](http://i.imgur.com/wwKmG2I.png)

**Comment by Yonezpt** [comment]
The feature stores the user setting in a cookie so if anyone has cookies disabled/blocked then the autoplay off will never work (unless on the first time) because it tries to find the user choice and defaults to on when it doesn't find it.

It is also possible that some times the cookie bugs out and stores an incorrect setting that is incompatible with the feature and so it again defaults to on.

I will try later to create a small script to remove that autoplay feature for anyone that wants it until YePpHa decides wether or not he will include this in YTC.

**Comment by WhyTheFuckAreAllNamesAlreadyTaken** [comment]
I have a suggestion..... Up Next is generally a good feature, only annoying fact is, YT gets to chose the video for u. What would be nice, is a function in YTC where i can click on related videos while the current video is playing, which are then stored in kind of an extra-list. As soon as the current video ends, the next video of all I have chosen would lo0ad. Alternatively, I could load the next one with lets say CTRL+right arrow or whatever. (Not the watch-later list on YT, where u need to go to that list and then play it.) That would be pretty epic, caus I often see many interesting related vids when watching, but being at the next one, I forgot the others i wanted to check. Then again, I need to go back, and forget the related ones on the new vid which i was interested in.... well yeah..... that would be epic..... ^^

**Comment by ErdeFB** [comment]
@WhyTheFuckAreAllNamesAlreadyTaken Please open a new ticket for your suggestion instead of posting it to a irrelevant existing issue report so it can actually be found later on.

**Comment by Yonezpt** [comment]
Alright, here it is the autoplay up next remover userscript, all you have to do is have a userscript addon/extensio manager installed (like Greasemonkey or Tampermonkey) and click on the "Raw" button, it should ask you if you want to install it and that's it, autoplay up next is gone.

https://gist.github.com/Yonezpt/51adf278a24488f75ff0

Works for Firefox and Chrome, let me know if it worked ok.

**Comment by ackern** [comment]
Marry me! That rocks!

Also it has the effect of making the thing that happens at the end of the video much quicker. The grid of suggested video thumbnails in the player window - it appears instantly now. Before (before Up Next existed), it loaded a lot slower which was hugely irritating. I think the thumbnails would load one by one, one after the other. Or they would fade in gradually. Can't remember exactly but it was stupid and annoying. Now, your video ends and bam the thumbnails are there immediately.

**Comment by z235** [comment]
@Yonezpt

Works like a charm. Thank you very much!

**Comment by Gittyperson** [comment]
Thanks very much Yonezpt for your script, works fine on Pale Moon. I really wish it'll be included in YTC, Up Next is extremely annoying.

**Comment by guiaurelio** [comment]
@Yonezpt Thanks !!!!

**Comment by Gittyperson** [comment]
Yonezpt's script not working anymore?

**Comment by Yonezpt** [comment]
@Gittyperson The script is still working for me on both Firefox and Chrome. Could you please discuss this in the script page instead of here? Otherwise we are abusing an issue page that is related to YTC, not to my userscript.

**Comment by Gittyperson** [comment]
You are right, I'm deleting both comments.

**Comment by timnyi** [comment]
I have this issue as well. Will be using @Yonezpt 's script for now

Using Firefox 38 Beta, Developer Version #501 with Greasemonkey.

**Comment by bb010g** [comment]
:+1: for @Yonezpt's script being put into YTC.

**Comment by guiaurelio** [comment]
I agree !!

**Comment by Consumered** [comment]
Very cool that there is a script for this. Will definitely appreciate if it is implemented directly with YouTube Center.

@Zarggg's comment about "default state" is incredibly fallacious when YouTube Center already has numerous options beyond extending YouTube by changing its default state (**center**, language, comment/description/share expansion, player, player resolution and size, and so on) that works with or without cookies. If it can be feasibly done, then "live with it" isn't a valid response given the nature of this extension.
